neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -179,7 +185,12 @@ bool IInventory.Verify(Snapshot snapshot)
 
         public virtual bool Verify(Snapshot snapshot, IEnumerable<Transaction> mempool)
         {
-            if (Size > MaxTransactionSize) return false;
+            int size = Size;
+            if (size > MaxTransactionSize) return false;
+            if (size > NativeContract.Policy.GetMaxLowPriorityTransactionSize(snapshot) && NetworkFee / size < NativeContract.Policy.GetFeePerByte(snapshot))
```

Comment:
Will be great if `Policy` calls will be cached only for verification.

Reply:
`Snapshot` is a cache.

Reply:
My fear is...

Verification of 100 tx with the same snapshot.

-> Read storage only one time for read the variable
-> Speed up the verification of the 2.. to `n` transactions

Application

-> If we cache the previous value, it could be return undesired values if we read and set in the same snapshot

Reply:
I didn't get it. How does it return undesired values?

Reply:
With 3 transactions

- The first read (and cache it) 
- The second write
- The third read again (the first cached value, not the modified)

Reply:
Currently, is not cached like 

```csharp
void GetMaxLowPriorityTransactionSize()
{
   if (snapshot.CachedGetMaxLowPriorityTransactionSize.HasValue) ...
   ...
}
```

So there are no problems, but not speed optimization too

Reply:
Reads in the same snapshot are optimized.